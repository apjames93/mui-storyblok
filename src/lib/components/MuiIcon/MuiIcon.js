import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import Icon from '@material-ui/core/Icon';
import Storyblok from '../../utils/Storyblok';
import { muiStringProp } from '../../utils/customProps';
import BBBSvg from './components/BBBSvg/BBBSvg';
import FacebookSvg from './components/FacebookSvg/FacebookSvg';
import GoogleSvg from './components/GoogleSvg/GoogleSvg';
import LinkedinSvg from './components/LinkedinSvg/LinkedinSvg';
// import styles for icons in styleguide
import '../../styles/styles.scss';

// import icons to use for custom icons passed down to app;
import customIcons from '../../utils/customIcons';

// import { icons } from '../MuiStoryblok/MuiStoryblok';
// const icons = getIcons();
/**
 * MuiIcon is used in storyblok redirect to react routes
 */

export const MuiIcon = ({
  color,
  fontSize,
  iconName,
  rootClass,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  const customIcon = customIcons.find(icon => icon.iconName === iconName);
  if (customIcon) {
    const Custom = customIcon.Component;
    return (
      <Custom
        className={styles.root}
        color={color}
        fontSize={fontSize}
        {...customIcon.props}
      />
    );
  }

  switch (iconName) {
    case 'facebook':
      return <FacebookSvg color={color} fontSize={fontSize} className={styles.root} />;
    case 'bbb':
      return <BBBSvg color={color} fontSize={fontSize} className={styles.root} />;
    case 'linkedin':
      return <LinkedinSvg color={color} fontSize={fontSize} className={styles.root} />;
    case 'google':
      return <GoogleSvg color={color} fontSize={fontSize} className={styles.root} />;
    default:
      return (
        <Icon
          className={styles.root}
          color={color}
          fontSize={fontSize}
        >
          {iconName}
        </Icon>
      );
  }
};

export default MuiIcon;

MuiIcon.propTypes = {
  /** any icon from https://material.io/resources/icons/?style=baseline */
  iconName: PropTypes.string.isRequired,
  /**
   * mui prop: 'inherit', 'primary', 'secondary', 'action', 'error', 'disabled'
   * The color of the component. It supports those theme colors that make sense for this component.
   * */
  color(props, propName, componentName) {
    const validProp = ['inherit', 'primary', 'secondary', 'action', 'error', 'disabled'];
    return muiStringProp(props, propName, componentName, validProp);
  },
  /**
   * mui prop: 'default', 'small', 'inherit', 'large'
   * The fontSize applied to the icon. Defaults to 24px, but can be configure to inherit font size.
   * */
  fontSize(props, propName, componentName) {
    const validProp = ['default', 'small', 'inherit', 'large'];
    return muiStringProp(props, propName, componentName, validProp);
  },
  /**
   * storyblok multiselect of css classes
   * Override or extend the styles applied to the component
   * */
  rootClass: PropTypes.arrayOf(PropTypes.string),
};

MuiIcon.defaultProps = {
  rootClass: [],
  color: 'secondary',
  fontSize: 'default',
};
