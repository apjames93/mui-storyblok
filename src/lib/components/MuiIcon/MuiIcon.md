MuiIcon example:
[View All MUI Icons](https://material.io/resources/icons/?style=baseline)

```js
  <div styles={{ display: 'flex', flexDirection: 'row' }}>
    <MuiIcon
      iconName='home'
      color='primary'
    />
    <MuiIcon
      iconName='menu'
      color='secondary'
    />
    <MuiIcon
      iconName='close'
      color='action'
    />
    <MuiIcon
      iconName='accessibility'
      color='inherit'
    />
    <MuiIcon
      iconName='done'
      color='error'
    />

      <MuiIcon
      iconName='done'
      color='error'
      fontSize="small"
    />
  </div>
```

Social Icons for MuiIcon examples:
* iconName = 'facebook'
* iconName = 'bbb'
* iconName = 'google'
* iconName = 'linkedin'
* iconName = 'youtube'
* iconName = 'instagram'
* iconName = 'twitter'
* iconName = 'yelp'
* iconName = 'github'

```js
  <div styles={{ display: 'flex', flexDirection: 'row' }}>
    <MuiIcon
      iconName='facebook'
      color='primary'
      fontSize='large'
    />
    <MuiIcon
      iconName='bbb'
      color='secondary'
      fontSize='large'
    />
    <MuiIcon
      iconName='google'
      color='action'
      fontSize='large'
    />
    <MuiIcon
      iconName='linkedin'
      color='inherit'
      fontSize='large'
      rootClass={['margin-left: 10px', 'margin-right: 10px']}
    />
    <MuiIcon
      iconName='youtube'
      color='inherit'
      fontSize='large'
    />
    <MuiIcon
      iconName='instagram'
      color='secondary'
      fontSize='large'
    />
    <MuiIcon
      iconName='twitter'
      color='primary'
      fontSize='large'
    />
    <MuiIcon
      iconName='yelp'
      color='secondary'
      fontSize='large'
    />
    <MuiIcon
      iconName='github'
      color='inherit'
      fontSize='large'
    />
  </div>
```