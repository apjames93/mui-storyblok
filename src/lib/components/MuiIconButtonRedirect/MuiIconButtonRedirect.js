import React from 'react';
import PropTypes from 'prop-types';
import MuiIconButton from '../MuiIconButton/MuiIconButton';
import { validComponentsRequired } from '../../utils/customProps';
import appRedirect from '../../utils/appRedirect';

/**
 * MuiIconButtonRedirect is used in storyblok redirect to react routes
 */

export const MuiIconButtonRedirect = ({
  redirectRoute,
  iconButton,
  callback,
}) => {
  const onClick = () => {
    if (callback) callback();
    appRedirect(redirectRoute);
  };

  const muiIconButton = iconButton[0];

  return (
    <MuiIconButton
      {...muiIconButton}
      onClick={onClick}
    />
  );
};

export default MuiIconButtonRedirect;

MuiIconButtonRedirect.propTypes = {
  /** redirect route */
  redirectRoute: PropTypes.string.isRequired,
  /** MuiIconButton Allowed maximum: 1 */
  iconButton(props, propName, componentName) {
    return validComponentsRequired(props, propName, componentName, ['MuiIconButton'], 1);
  },
  callback: PropTypes.func,
};

MuiIconButtonRedirect.defaultProps = {
  iconButton: [],
  callback: undefined,
};
