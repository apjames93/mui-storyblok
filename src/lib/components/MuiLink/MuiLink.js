import React from 'react';
import PropTypes from 'prop-types';
import Link from '@material-ui/core/Link';
import Storyblok from '../../utils/Storyblok';

/**
 * MuiLink is used in storyblok redirect to a href
 */

export const MuiLink = ({
  text,
  color,
  href,
  underline,
  rootClass,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  const onClick = () => {
    window.location.replace(href);
  };

  return (
    // eslint-disable-next-line jsx-a11y/anchor-is-valid
    <Link
      className={styles.root}
      color={color}
      underline={underline}
      href="#"
      onClick={onClick}
    >
      {text}
    </Link>
  );
};

export default MuiLink;

MuiLink.propTypes = {
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui prop:
   'initial' | 'inherit' | 'primary' | 'secondary' | 'textPrimary' | 'textSecondary' | 'error'
   * The color of the component. It supports those theme colors that make sense for this component.
   * */
  color: PropTypes.string,
  /*
  * mui prop:'none' | 'hover' | 'always'
  * underline text
  */
  underline: PropTypes.string,
  /* url for redirect */
  href: PropTypes.string.isRequired,
  /* text to click on */
  text: PropTypes.string.isRequired,
};

MuiLink.defaultProps = {
  color: 'primary',
  underline: 'hover',
  rootClass: [],
};
