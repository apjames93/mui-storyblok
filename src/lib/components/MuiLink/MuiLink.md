MuiLink Examples:

```js
const underline = 'hover';
const href = 'https://github.com/apjames93/mui-storyblok';
const text = 'MuiStoryblok Github';

<MuiLink
  underline={underline}
  href={href}
  text={text}
/>

```