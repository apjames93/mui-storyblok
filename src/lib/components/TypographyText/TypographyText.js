import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import Storyblok from '../../utils/Storyblok';

/**
 * TypographyText
 */

export const TypographyText = ({
  align,
  color,
  height,
  width,
  rootClass,
  content,
  components,
}) => {
  const defaultStyles = {
    height,
    width,
  };
  const styles = Storyblok.arrayToMuiStyles(rootClass, defaultStyles);

  return (
    <Typography
      className={styles.root}
      align={align}
      color={color}
      component="div"
    >
      {content.map((item, index) => {
        if (item.component === 'MuiText') {
          const style = Storyblok.arrayToMuiStyles(item.rootClass);
          return (
            <Typography
              component="span"
              key={index}
              className={style.root}
              variant={item.variant}
              align={item.align}
              color={item.color}
            >
              {` ${item.text} `}
            </Typography>
          );
        }
        return createElement(components[item.component], Object.assign(item, { key: index }));
      })}
    </Typography>
  );
};

export default TypographyText;

TypographyText.propTypes = {
  /**
   * storyblok multiselect of css classes
   * Override or extend the styles applied to the component
   * */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * mui prop: 'inherit' | 'left' | 'center' | 'right' | 'justify'
   * Set the text-align on the component.
   * */
  align: PropTypes.string,
  /**
   * mui prop: 'initial'| 'inherit'| 'primary'| 'secondary'| 'textPrimary'| 'textSecondary'| 'error'
   * The color of the component. It supports those theme colors that make sense for this component.
   * */
  color: PropTypes.string,
  /** hight of the Typography element */
  height: PropTypes.string,
  /** width of the Typography element */
  width: PropTypes.string,

  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,

  /** componets to render in Typography */
  components: PropTypes.shape(PropTypes.elementType).isRequired,
};

TypographyText.defaultProps = {
  width: '100%',
  height: '100%',
  align: 'inherit',
  color: 'initial',
  rootClass: [],
};
