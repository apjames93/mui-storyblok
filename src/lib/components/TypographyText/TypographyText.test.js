import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import TypographyText from './TypographyText';
import MuiIconButtonRedirect from '../MuiIconButtonRedirect/MuiIconButtonRedirect';
import MuiLink from '../MuiLink/MuiLink';
import MuiIcon from '../MuiIcon/MuiIcon';

function setup() {
  const props = {
    components: {
      MuiIconButtonRedirect,
      MuiLink,
      MuiIcon,
    },
    content: [{
      component: 'MuiText',
      text: 'this is text',
    }, {
      component: 'MuiLink',
      text: 'btn',
      href: 'google.com',
    }, {
      component: 'MuiIconButtonRedirect',
      redirectRoute: 'page-welcom',
      iconButton: [{
        component: 'MuiIconButton',
        icon: [{
          component: 'MuiIcon',
          iconName: 'star',
        }],
      }],
    }, {
      component: 'MuiIcon',
      iconName: 'android',
    }],
  };
  const comp = shallow(<TypographyText {...props} />);
  return { comp, props };
}

describe('<TypographyText />', () => {
  it('renders TypographyText', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create((
      <MemoryRouter>
        <TypographyText {...props} />
      </MemoryRouter>
    ));
    expect(tree).toMatchSnapshot();
  });
});
