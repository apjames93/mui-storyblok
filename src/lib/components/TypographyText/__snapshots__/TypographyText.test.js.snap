// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypographyText /> snapshot 1`] = `
<div
  className="MuiTypography-root makeStyles-root-5 MuiTypography-body1"
>
  <span
    className="MuiTypography-root makeStyles-root-6 MuiTypography-body1"
  >
     this is text 
  </span>
  <a
    className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-root-7 MuiTypography-colorPrimary"
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
  >
    btn
  </a>
  <button
    className="MuiButtonBase-root MuiIconButton-root makeStyles-root-8"
    data-testid="muiIconButton"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root makeStyles-root-9 MuiIcon-colorSecondary"
      >
        star
      </span>
    </span>
  </button>
  <span
    aria-hidden={true}
    className="material-icons MuiIcon-root makeStyles-root-10 MuiIcon-colorSecondary"
  >
    android
  </span>
</div>
`;
